src/neo/Ledger/StorageKey.cs

```diff
@@ -10,31 +11,22 @@ public class StorageKey : IEquatable<StorageKey>, ISerializable
         public int Id;
         public byte[] Key;
 
-        int ISerializable.Size => sizeof(int) + (Key.Length / 16 + 1) * 17;
+        int ISerializable.Size => sizeof(int) + Key.Length;
 
-        internal static byte[] CreateSearchPrefix(int id, byte[] prefix)
+        internal static byte[] CreateSearchPrefix(int id, ReadOnlySpan<byte> prefix)
         {
-            using (MemoryStream ms = new MemoryStream())
-            {
-                int index = 0;
-                int remain = prefix.Length;
-                while (remain >= 16)
-                {
-                    ms.Write(prefix, index, 16);
-                    ms.WriteByte(16);
-                    index += 16;
-                    remain -= 16;
-                }
-                if (remain > 0)
-                    ms.Write(prefix, index, remain);
-                return Helper.Concat(BitConverter.GetBytes(id), ms.ToArray());
-            }
+            byte[] buffer = new byte[sizeof(int) + prefix.Length];
+            BinaryPrimitives.WriteInt32LittleEndian(buffer, id);
+            prefix.CopyTo(buffer.AsSpan(sizeof(int)));
+            return buffer;
         }
 
+        //If the base stream of the reader doesn't support seeking, a NotSupportedException is thrown.
+        //But StorageKey never works with NetworkStream, so it doesn't matter.
         void ISerializable.Deserialize(BinaryReader reader)
         {
             Id = reader.ReadInt32();
-            Key = reader.ReadBytesWithGrouping();
+            Key = reader.ReadBytes((int)(reader.BaseStream.Length - reader.BaseStream.Position));
```

Comment:
if we have a max key of 64, and the counter is 4 bytes (integer), we can read this length into a buffer, and trim according to the `Read` return, in this way, we don't need to check the `Length` of the stream.

Reply:
If it is a network stream, you will consume the next message. If it isn't a network stream, we can read the length.